### PR TITLE
Make CI switch branches to PR ref so we don't just test master all the time 🙃

### LINKED ISSUE #994: Status checks are passing, but locally, tests are failing
it seems that CI is not testing the right code from PRs since this is happening:
https://github.com/fastlane/ci/issues/993 (Broken tests for settings)
But the PR that merged these things was green: https://github.com/fastlane/ci/pull/935

### PR DESCRIPTION
**Pull request by taquitos**
Fixes #994
removes Extra work being done on checkout

- [x] I have run `rspec` and corrected all errors
- [x] I have run `rubocop` and corrected all errors
- [x] I have tested this change locally and tried to launch the server as well as access a project, and with that at least one build
- [x] If there is an existing issue, make sure to add `Fixes ...` as part of the PR body to reference the issue you're solving

### CODE REVIEW

File: app/features/project/project_controller.rb [hunk 1]
```diff
@@ -34,14 +34,20 @@ class ProjectController < AuthenticatedControllerBase
       else
         checkout_folder = File.join(File.expand_path(project.local_repo_path), "manual_build_#{sha_or_uuid}")
       end
+
       # TODO: This should be hidden in a service
-      repo = FastlaneCI::GitRepo.new(
```
Comment (taquitos) [inline app/features/project/project_controller.rb]: this was extra work that was being done, often for no reason.

File: app/features/build_runner/fastlane_build_runner.rb [hunk 2]
```diff
@@ -44,11 +44,10 @@ def setup(parameters: nil)
       # and we can read files like the `xcode-version` file
       current_build.build_tools = {}
 
-      # The call below could be optimized, as it will also set the status
-      # on the GitHub remote. We want to store the lane, platform and parameters
-      # that's why we call it here in the `setup` method again, as well as in
-      # `BuildRunner#prepare_build_object`
-      save_build_status!
```
Comment (taquitos) [inline app/features/build_runner/fastlane_build_runner.rb]: don't set github status 2 times to pending 🎉

File: app/features/build_runner/build_runner.rb [hunk 3]
```diff
@@ -78,10 +89,11 @@ def initialize(project:, sha:, github_service:, notification_service:, work_queu
 
       prepare_build_object(trigger: trigger)
 
+      local_folder = @local_build_folder || File.join(project.local_repo_path, "builds", sha)
       @repo = GitRepo.new(
         git_config: project.repo_config,
         provider_credential: github_service.provider_credential,
-        local_folder: File.join(project.local_repo_path, "builds", sha),
```
Comment (taquitos) [inline app/features/build_runner/build_runner.rb]: don't always assume we're using the `builds` folder, we might have a 1-off build

File: app/shared/models/git_repo.rb [hunk 4]
```diff
@@ -472,7 +472,7 @@ def perform_block(use_global_git_mutex: true, &block)
     end
 
     def pull(repo_auth: self.repo_auth, use_global_git_mutex: true)
-      logger.debug("Enqueuing a pull on `master` (with mutex?: #{use_global_git_mutex}) for #{git_config.git_url}")
```
Comment (taquitos) [inline app/shared/models/git_repo.rb]: it's not always `master`

File: app/shared/models/git_repo.rb [hunk 5]
```diff
@@ -641,7 +641,7 @@ def switch_to_git_fork(clone_url:, branch:, sha: nil, local_branch_name:, use_gl
         logger.debug("Switching to branch #{branch} from forked repo: #{clone_url} (pulling into #{local_branch_name})")
 
         begin
-          git.branch(local_branch_name)
+          git.branch(local_branch_name).checkout
```
Comment (taquitos) [inline app/shared/models/git_repo.rb]: whoopsies

File: app/shared/models/git_repo.rb [hunk 6]
```diff
@@ -668,6 +668,7 @@ def switch_to_ref(git_fork_config:, local_branch_name:, use_global_git_mutex: fa
           logger.debug("Switching to new branch from ref #{ref} (pulling into #{local_branch_name})")
           git.fetch(GitRepo::DEFAULT_REMOTE, ref, {})
           git.branch(local_branch_name)
+          git.checkout(local_branch_name)
```
Comment (taquitos) [inline app/shared/models/git_repo.rb]: whoopsies

File: launch.rb [hunk 7]
```diff
@@ -254,6 +254,8 @@ def self.run_pending_github_builds(projects: nil, github_service: nil)
             ref: matching_open_pr.git_ref
           )
 
+          # TODO: should make sure we don't already have a checkout, if we do, we need to adjust
```
Comment (taquitos) [inline launch.rb]: probably not a big deal, but good to deal with this corner-case

File: app/features/build_runner/build_runner.rb [hunk 8]
```diff
@@ -50,11 +50,21 @@ class BuildRunner
     # Array of env variables that were set, that we need to unset after the run
     attr_accessor :environment_variables_set
 
-    def initialize(project:, sha:, github_service:, notification_service:, work_queue:, trigger:, git_fork_config:)
+    attr_accessor :local_build_folder
```
Comment (KrauseFx) [inline app/features/build_runner/build_runner.rb]: Would be good to have a comment here on what the attribute is used for

File: app/shared/models/git_repo.rb (hunk 5 above)
Comment (KrauseFx) [reply app/shared/models/git_repo.rb]: 👍

**Comment by KrauseFx** [comment]
Let's merge this as soon as you're online :)

**Pull request merged**